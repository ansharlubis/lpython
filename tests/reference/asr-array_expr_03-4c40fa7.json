{
    "basename": "asr-array_expr_03-4c40fa7",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/array_expr_03.py",
    "infile_hash": "780141378ce8b8bc3a2bcb3769c12d720dc7b7b990806a8e69fa3764",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-array_expr_03-4c40fa7.stdout",
    "stdout_hash": "f70c88e7da5a83ec73c838cfb99e90491ac22fe130ba1ff88f18d2fb",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}