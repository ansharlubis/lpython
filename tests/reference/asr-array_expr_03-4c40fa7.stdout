(TranslationUnit (SymbolTable 1 {f: (Function (SymbolTable 2 {_lpython_return_variable: (Variable 2 _lpython_return_variable ReturnVar () () Default (Real 8 [((IntegerConstant 0 (Integer 4 [])) (IntegerConstant 1234 (Integer 4 [])))]) Source Public Required .false.), a: (Variable 2 a InOut () () Default (Real 8 [(() ())]) Source Public Required .false.), b: (Variable 2 b InOut () () Default (Real 8 [(() ())]) Source Public Required .false.), i: (Variable 2 i Local () () Default (Integer 4 []) Source Public Required .false.), r: (Variable 2 r Local () () Default (Real 8 [((IntegerConstant 0 (Integer 4 [])) (IntegerConstant 1234 (Integer 4 [])))]) Source Public Required .false.)}) f [(Var 2 a) (Var 2 b)] [(DoLoop ((Var 2 i) (IntegerConstant 0 (Integer 4 [])) (IntegerBinOp (IntegerConstant 1234 (Integer 4 [])) Sub (IntegerConstant 1 (Integer 4 [])) (Integer 4 []) (IntegerConstant 1233 (Integer 4 []))) (IntegerConstant 1 (Integer 4 []))) [(= (ArrayItem (Var 2 r) [(() (Var 2 i) ())] (Real 8 []) ()) (RealBinOp (ArrayItem (Var 2 a) [(() (Var 2 i) ())] (Real 8 []) ()) Add (ArrayItem (Var 2 b) [(() (Var 2 i) ())] (Real 8 []) ()) (Real 8 []) ()) ())]) (= (Var 2 _lpython_return_variable) (Var 2 r) ()) (Return)] (Var 2 _lpython_return_variable) Source Public Implementation () .false. .false. .false. .false. [] [] .false.), main: (Function (SymbolTable 3 {__lcompilers_dummy: (Variable 3 __lcompilers_dummy Local () () Default (Real 8 [((IntegerConstant 0 (Integer 4 [])) (IntegerConstant 1234 (Integer 4 [])))]) Source Public Required .false.), a: (Variable 3 a Local () () Default (Real 8 [((IntegerConstant 0 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])))]) Source Public Required .false.), b: (Variable 3 b Local () () Default (Real 8 [((IntegerConstant 0 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])))]) Source Public Required .false.)}) main [] [(= (Var 3 __lcompilers_dummy) (FunctionCall 1 f () [((Var 3 a)) ((Var 3 b))] (Real 8 [((IntegerConstant 0 (Integer 4 [])) (IntegerConstant 1234 (Integer 4 [])))]) () ()) ())] () Source Public Implementation () .false. .false. .false. .false. [] [] .false.), main_program: (Program (SymbolTable 5 {}) main_program [] [])}) [])
